src/topology/separation.lean

```diff
@@ -150,6 +150,29 @@ instance subtype.t1_space {α : Type u} [topological_space α] [t1_space α] {p
 instance t1_space.t0_space [t1_space α] : t0_space α :=
 ⟨λ x y h, ⟨{z | z ≠ y}, is_open_ne, or.inl ⟨h, not_not_intro rfl⟩⟩⟩
 
+lemma t1_characterisation : t1_space α ↔
+  ∀ (x y), x ≠ y → (∃ (U : set α) (hU : is_open U), x ∈ U ∧ y ∉ U) :=
+begin
+  split;intro h,
+  { intros x y hxy,
+    exact ⟨{y}ᶜ,is_open_compl_iff.mpr (@t1_space.t1 α _inst_1 h y), mem_compl_singleton_iff.mpr hxy,
+      not_not.mpr rfl⟩},
+  { fconstructor,
+    intro x,
+    fconstructor,
+    have p : ⋃₀ {U : set α | (x ∉ U) ∧ (is_open U)} = {x}ᶜ,
+    { apply subset.antisymm; intros t ht,
+      { rcases ht with ⟨A,⟨hxA, hA⟩, htA⟩,
+        simp only [mem_singleton_iff, mem_compl_eq],
+        intro htx,
+        simpa [*] using htA},
+      { obtain ⟨U, hU, hh⟩ := h t x (mem_compl_singleton_iff.mp ht),
+        exact ⟨U, ⟨hh.2, hU⟩, hh.1⟩}},
+    rw ← p,
+    have c : ∀ B ∈ {U : set α | (x ∉ U) ∧ (is_open U)}, is_open B, by finish,
```

Comment:
```suggestion
    have c : ∀ B ∈ {U : set α | (x ∉ U) ∧ (is_open U)}, is_open B, by simp,
```

I'm not sure about this — I thought `finish` was a heavier tactic we're meant to avoid when possible, but unsure. You could even do this as `apply is_open_sUnion, simp`

Reply:
(which saves you from typing out the long type)

Reply:
```suggestion
    exact is_open_sUnion (λ B hB, hB.2)}
```
I think the line 172 it's no necessary, insteed we can writte the exact with the proof of `c` statment. 
It's my first PR, sorry about the chaos. I don't know how to make suggestions.